src/main/java/org/apache/sling/discovery/impl/DiscoveryServiceConfig.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.sling.discovery.impl;
+
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+@ObjectClassDefinition(
+        localization = "OSGI-INF/l10n/metatype",
+        name = "%config.name",
+        description = "%config.description"
+)
+public @interface DiscoveryServiceConfig {
```

Comment:
If you care about backwards compatibility of the configuration, you should add `pid = "org.apache.sling.discovery.impl.Config"` to the annotation.

Reply:
The discovery.oak bundle has a similar structure and went through the same migration to official OSGi annotations. The config class there doesn't have a pid. I'm also a bit confused, because I understand the configuration pid is defined on `org.apache.sling.discovery.impl.Config` with Component(configurationPid=...) and the value defaults to the name of the component, which itself defaults to the fully qualified class name.